hadoop-ozone/dist/src/main/compose/ozone-mr/hadoop31/test.sh

```diff
@@ -23,12 +23,10 @@ source "$COMPOSE_DIR/../../testlib.sh"
 
 start_docker_env
 
-execute_robot_test scm createmrenv.robot
-
-
 #rm is the container name (resource manager) and not the rm command
-execute_command_in_container rm sudo apk add --update py-pip
-execute_command_in_container rm sudo pip install robotframework
+execute_command_in_container rm bash -c "if test -e /sbin/apk; then sudo apk add --update py-pip; sudo pip install robotframework; fi"
```

Comment:
Do we need to add this check for hadoop27/test.sh as well?

When /sbin/apk does not exist, we skip pip install, etc via apk. How does the test pass without a robotframework install in the new image? If the new image moving forward already contain the roboframework, can we remove the logic?

should we use "-f" instead of "-e" to test the existence of a file?

Reply:
> Do we need to add this check for hadoop27/test.sh as well?

The docker image for 2.7 hasn't been updated to CentOS, so not needed there.

> When /sbin/apk does not exist, we skip pip install, etc via apk. How does the test pass without a robotframework install in the new image? If the new image moving forward already contain the roboframework, can we remove the logic?

Yes, the new image has both `pip` and `robotframework`.  The conditional logic to install them can be safely removed, as long as Alpine is gone for good.

> should we use "-f" instead of "-e" to test the existence of a file?

My intention was to make it work with symlinks.  But after some testing it turns out both `-f` and `-e` return true for links that point to files, too.  So `-f` is probably better, but it only matters in case `/sbin/apk` happens to be a dir.